### PR TITLE
Day six

### PR DESCRIPTION
**Pull request by Qairkhan**
Удален файл render.js для избавления зависимости. Добавлены observer и subscribe. Также удален лишний файл moks.js (бел преобразован в state.js)

### CODE REVIEW

File: src/components/Profile/Profile.jsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ const Profile = (props) => {
   return (
     <div>
       <ProfileInfo />
-      <MyPosts posts={props.posts} />
+      <MyPosts posts={props.posts} addPost={props.addPost} newPostText={props.posts.newPostText} updateNewPostText={props.updateNewPostText}/>
```
Comment (Dmitry-White) [inline src/components/Profile/Profile.jsx]: Whenever you've got more than 3 props, consider formatting you JSX like this:
image

File: src/components/redux/State.js [hunk 2]
```diff
@@ -30,13 +29,28 @@ const state = {
...
+
+  const updateNewpostText = (newText) => {
+    state.ProfilePage.newPostText = newText;
+    rerenderEntireTree(state);
+  }
+
+  const subscribe = (observer) => {
+    rerenderEntireTree = observer;
+
+  }
```
Comment (Dmitry-White) [inline src/components/redux/State.js]: Consider adding an empty line after the last "}".

File: src/components/redux/State.js [hunk 3]
```diff
@@ -30,13 +29,28 @@ const state = {
...
     rerenderEntireTree(state);
   };
-export {state,addPost};
\ No newline at end of file
+
+let rerenderEntireTree = () => {
+
+}
+
+  const updateNewpostText = (newText) => {
```
Comment (Dmitry-White) [inline src/components/redux/State.js]: Consider fixing the indentation here - line 47 should not start with a space.

File: src/index.js [hunk 4]
```diff
@@ -1,15 +1,24 @@
...
-import {rerenderEntireTree} from './render'
-import {state} from './components/redux/State'
+import {state, addPost, updateNewpostText, subscribe} from './components/redux/State'
+
+const rerenderEntireTree = (state) => {
+ReactDOM.render(
+  <BrowserRouter>
+  <React.StrictMode>
+    <App appState={state} addPost={addPost} updateNewPostText={updateNewpostText}/>
+  </React.StrictMode>,
```
Comment (Dmitry-White) [inline src/index.js]: The comma "," after </React.StrictMode> is redundant.

File: src/index.js [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
+import ReactDOM from 'react-dom';
+import {BrowserRouter, Route} from "react-router-dom"
+import './index.css';
+import App from './App';
 import * as serviceWorker from './serviceWorker';
-import {rerenderEntireTree} from './render'
-import {state} from './components/redux/State'
+import {state, addPost, updateNewpostText, subscribe} from './components/redux/State'
+
+const rerenderEntireTree = (state) => {
```
Comment (Dmitry-White) [inline src/index.js]: Consider fixing the indentation like this:
image

File: src/components/redux/State.js [hunk 6]
```diff
@@ -30,13 +29,28 @@ const state = {
...
+    state.ProfilePage.newPostText = newText;
+    rerenderEntireTree(state);
+  }
+
+  const subscribe = (observer) => {
+    rerenderEntireTree = observer;
+
```
Comment (Dmitry-White) [inline src/components/redux/State.js:55]: The empty line is redundant here.

**Review by Dmitry-White — CHANGES_REQUESTED** [review]

File: src/components/Profile/Profile.jsx (hunk 1 above)
Comment (Qairkhan) [reply src/components/Profile/Profile.jsx]: done.

File: src/components/redux/State.js (hunk 2 above)
Comment (Qairkhan) [reply src/components/redux/State.js]: done

File: src/components/redux/State.js (hunk 3 above)
Comment (Qairkhan) [reply src/components/redux/State.js]: done

File: src/index.js (hunk 4 above)
Comment (Qairkhan) [reply src/index.js]: done

File: src/index.js (hunk 5 above)
Comment (Qairkhan) [reply src/index.js]: done

File: src/components/redux/State.js (hunk 6 above)
Comment (Qairkhan) [reply src/components/redux/State.js:55]: done

File: src/index.js [hunk 7]
```diff
@@ -1,15 +1,24 @@
-
-
 import './index.css';
-
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {BrowserRouter, Route} from "react-router-dom"
+import './index.css';
```
Comment (Dmitry-White) [inline src/index.js]: Consider following the already mentioned import guidelines.
Group the imports by purpose. First go third party imports, last - css.

**Review by Dmitry-White — CHANGES_REQUESTED** [review]

**Review by Dmitry-White — APPROVED** [review]

**Pull request merged**